Documentation/Evolution/StronglyTypedCaptures.md

```diff
@@ -0,0 +1,560 @@
+    - [Alternatives considered](#alternatives-considered)
+    - [Future directions](#future-directions)
+        - [Dynamic captures](#dynamic-captures)
+
+<!-- markdown-toc end -->
+
+## Introduction
+
+Capturing groups are a commonly used component of regular expressions as they
+allow the programmer to extract information from matched input. A capturing
+group groups multiple characters together as a single unit that can be
+[backreferenced](https://www.regular-expressions.info/backref.html) within the
+regular expression and accessed in the result of a successful match. For
+example, the following regular expression contains the capturing groups `(cd*)`
+and `(ef)`.
+
+```swift
+// Literal version.
+let regex = /ab(cd*)(ef)gh/
+// => `Regex<(Substring, Substring)>`
+
+// Result builder DSL equivalent:
+//     let regex = Regex {
+//         "ab"
+//         Regex {
+//             "c"
+//             Repeat("d")
+//         }.capture()
+//         "ef".capture()
+//         "gh"
+//     }
+
+if let match = "abcddddefgh".match(regex) {
+    print(match.captures) // => ("cdddd", "ef")
+}
+```
+
+We introduce a generic type `Regex<Captures>`, which treats the type of captures
+as part of a regular expression's type information for type safety and ease of
+use. As we explore a fundamental design aspect of the regular expression
+feature, this pitch discusses the following topics.
+
+- A type definition of generic type `Regex<Captures>`.
+- Capture type inference and composition in regular expression literals and the
+  forthcoming result builder DSL.
+- New language features which this design may require.
+
+This focus of this pitch is the structural properties of capture types and how
+regular expression patterns compose to form new capture types. The semantics of
+string matching, its effect on the capture types (i.e. `UnicodeScalarView` or
+`Substring`), or the matching API will be discussed in a separate pitch.
+
+For background on Declarative String Processing, see related topics:
+- [Declarative String Processing Overview](https://forums.swift.org/t/declarative-string-processing-overview/52459)
+- [Regular Expression Literals](https://forums.swift.org/t/pitch-regular-expression-literals/52820)
+- [Character Classes for String Processing](https://forums.swift.org/t/pitch-character-classes-for-string-processing/52920)
+
+## Motivation
+
+Across a variety of programming languages, many established regular expression
+libraries present captures as a collection of captured content to the caller
+upon a successful match
+[[1](https://developer.apple.com/documentation/foundation/nsregularexpression)][[2](https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.capture)].
+However, to know the structure of captured contents, programmers often need to
+read the regular expression, which may be a literal or a series of API calls, or
+run the regular expression on some input to find out. Because regular
```

Comment:
I found this sentence a little hard to parse. Maybe:
```suggestion
However, to know the structure of captured contents, programmers often need to
carefully read the regular expression or run the regular expression on some
input to find out. Because regular
```